queries.sql

```diff
@@ -113,3 +113,82 @@ join owners on owners.id = animals.owner_id
 group by owners.full_name 
 order by maximum 
 DESC LIMIT 1;
+
+-- Who was the last animal seen by William Tatcher?
+
+SELECT animals.name
+FROM visits 
+JOIN animals on animals.id = visits.animals_id
+JOIN vets on vets.id = visits.vets_id
+WHERE vets.name = 'William Tatcher'
+ORDER BY visits_date 
+DESC LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+
+SELECT vets.name as vets_name, count(*)
+FROM visits
+JOIN animals on animals.id = visits.animals_id
+JOIN vets on vets.id = visits.vets_id 
+WHERE vets.name = 'Stephanie Mendez'
+GROUP BY vets.name
+
+-- List all vets and their specialties, including vets with no specialties.
+
+SELECT vets.name as vets_name, species.name as species_name
+FROM vets
+LEFT JOIN specializations on vets.id = specializations.vets_id
+LEFT JOIN species on species.id = specializations.species_id
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+
+SELECT animals.name
+FROM visits
+JOIN vets on vets.id = visits.vets_id
+JOIN animals on animals.id = visits.animals_id
+WHERE vets.name = 'Stephanie Mendez' 
+and visits_date between 'April-01-2020' and 'August-30-2020'
+
+-- What animal has the most visits to vets?
+
+SELECT animals.name, count(*) as num_of_times
+FROM visits
+JOIN vets on vets.id = visits.vets_id
+JOIN animals on animals.id = visits.animals_id
+group by animals.name
+order by num_of_times desc limit 1
+
+-- Who was Maisy Smith's first visit?
+
+SELECT animals.name
+FROM visits
+JOIN vets on vets.id = visits.vets_id
+JOIN animals on animals.id = visits.animals_id
+where vets.name = 'Maisy Smith'
+order by visits_date
+limit 1
+
+-- Details for most recent visit: animal information, vet information, and date of visit.
+
+SELECT animals.name, vets.name, visits_date
+FROM visits
+JOIN vets on vets.id = visits.vets_id
+JOIN animals on animals.id = visits.animals_id
+order by visits_date desc limit 1
+
+-- How many visits were with a vet that did not specialize in that animal's species?
+
+SELECT count(*)
+FROM vets
+left JOIN visits on vets.id = visits.vets_id
+left JOIN specializations on vets.id = specializations.vets_id
+left JOIN species on species.id = specializations.species_id
+where species.name is null
+
+-- What specialty should Maisy Smith consider getting? Look for the species she gets the most.
+
+SELECT count(*)
+FROM vets
+JOIN specializations on vets.id = specializations.vets_id
+JOIN species on species.id = specializations.species_id
+WHERE vets.name = 'Maisy Smith'
```

Comment:
- [ ] The data returned for this query is wrong. It returns 0 while it should be `Digimon 6` based on the visits that Maisy Smith had. Kindly change the query and update the screenshot. 👍

Reply:
@Fayob, the query should return visits to vets by animals but those vets do not specialize in the animal's species, There are 12 such visits in total. The breakdown is as follows:

- Maisy Smith - 9 visits
- William Thatcher - 2 visits and

- Jack Harkness - 1 visit

Reply:
The question is confusing I don't understand it. I've been trying to digest it to be able to write the query since but i still don't get it. It's actually the major reason I submitted for review late

Reply:
@Gambit142, I go through the data given, specialiazation is the table that connects vets and species, and maisy smith was not given any species. 

https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_join_table.md

Reply:
Yes, I am not saying that specialization is not the table to use, I am just saying that the correct data should return `12`. Remember that `Maisy Smith` doesn't specialize in any, so all animals that visited her will fall into this category. Also, `William Thatcher` who is a specialist in `Pokemon` had two visits from `Digimons` while `Jack Harkness` who is a specialist in `Digimons` had one visit from a `pokemon`

Reply:
@Gambit142, the query is for this question "What specialty should Maisy Smith consider getting? Look for the species she gets the most."  which seems to be the last the question. It is not related to visits, it seems you're mixing up the query and the screenshot.

Reply:
@Fayob, oh sorry yes I am mixing up your follow-up question here. However, my initial inline comment still explains the change to be made. This query should return the species of animals that visited **_Maisy Smith_** more. If you look at your project, **_Maisy Smith_** had visits from a total of 9 animals. `3 Pokemons` and `6 Digimons`. Therefore, the query should return `Digimon 6` because it seems **_Maisy Smith_** knows how to treat them better. 👍 I hope this is clearer.

Reply:
@Fayob, I am glad I could be of help. 👍